README.md
# Verve

A Flutter theming solution, building on modern Material and Scandinavian
 principles. This package contains prebuilt theme swatches, as well as 
 opinionated Widgets designed to work with provided themes.

### Swatches

To access Verve's inbuilt swatches, use the method 
`verveSwatch({@required VerveTheme theme, ThemeData base})`. See 
**Working Example** to see how to do this correctly.

This returns a ThemeData, which you can use as the `theme` attribute of MaterialApp.

The available `VerveTheme`s are detailed in **Themes** below.

When you call `verveSwatch()`, verveSwatch returns a modified version of 
`base` with modified fonts, colors, Widget styling, and brightness attributes. 
If `base` is null, it will return a ThemeData with just these attributes.

Verve uses the [Metropolis](https://github.com/chrismsimpson/Metropolis) 
font, due to its clean, geometric nature. This font is free of licenses
and is included in this package. Verve also updates the Flutter font
defaults to the Material Design 2018 specification, as Flutter uses
the 2014 version currently.

verveSwatch also edits some components (Buttons, Cards, FABs) slightly
to look nicer stylistically. These are somewhat opinionated, but you can
always override them.

### Themes
Verve has 11 Swatches, which can be used to auto-style your Flutter apps.
Each cherry-picks 3 colors from a popular (Adobe Color) palette and 
transports it to Flutter. You can see each of them below in their
Adobe palette form, and the specific `Color`s used are included.

For some swatches, you may have to manually edit text colors. I am looking
for a workaround to this...

##### Material (`VerveTheme.material`)
* Adobe Color: N/A, it's Material colors.
* Ships with Flutter, only typography changes made.
* Widgets and Colors are **not** styled automatically.
##### Dark Mode (`VerveTheme.darkMode`)
* Adobe Color: N/A, it's Material Dark colors.
* Ships with Flutter, only typography changes made.
* Widgets and Colors are **not** styled automatically.
* May look somewhat strange, since this is experimental. Use Twilight if you are having issues.
##### Navy (`VerveTheme.navy`)
* Adobe Color: [KnotJustNautical](https://color.adobe.com/KnotJustNautical-color-theme-2565165)
* `Color`s:
    - Main: `Color(0xff2C3E50)`
    - Accent: `Color(0xffFC4349)`
    - Background: `Color(0xffffffff)`
##### Blue (`VerveTheme.blue`)
* Adobe Color: [Water](https://color.adobe.com/water-color-theme-13663112)
* `Color`s:
    - Main: `Color(0xff2787B7)`
    - Accent: `Color(0xff024669)`
    - Background: `Color(0xffffffff)`
##### Purple (`VerveTheme.purple`)
* Adobe Color: [Tema de Cores](https://color.adobe.com/Tema-de-cores-75-color-theme-13709296)
* `Color`s:
    - Main: `Color(0xff73305B)`
    - Accent: `Color(0xff272640)`
    - Background: `Color(0xffffffff)`
##### Beach (`VerveTheme.beach`)
* Adobe Color: [Sandy Stone](https://color.adobe.com/sandy-stone-beach-ocean-diver-color-theme-15325)
* `Color`s:
    - Main: `Color(0xff046380)`
    - Accent: `Color(0xffE6E2AF)`
    - Background: `Color(0xffffffff)`
##### Garden (`VerveTheme.garden`)
* Adobe Color: [Summer Garden](https://color.adobe.com/Copy-of-summer-garden-color-theme-11113033)
* `Color`s:
    - Main: `Color(0xff4BB5C1)`
    - Accent: `Color(0xffB5E655)`
    - Background: `Color(0xffffffff)` 
##### Sepia (`VerveTheme.sepia`)
* Adobe Color: [Aviator](https://color.adobe.com/en/Aviator-color-theme-1126690)
* `Color`s:
    - Main: `Color(0xff1A9481)`
    - Accent: `Color(0xff9BCC93)`
    - Background: `Color(0xffffffff)` 
##### Twilight (`VerveTheme.twilight`)
* Adobe Color: [Quail Vista Night](https://color.adobe.com/Quail-Vista-Night-color-theme-13484149/)
* `Color`s:
    - Main: `Color(0xff192853)`
    - Accent: `Color(0xffC9D2D9)`
    - Background: `Color(0xff0C1A3D)` 
##### Amati (`VerveTheme.amati`)
* Adobe Color: [Madame Amati](https://color.adobe.com/Madam-Amati-color-theme-13663575)
* `Color`s:
    - Main: `Color(0xff45757D)`
    - Accent: `Color(0xff27455C)`
    - Background: `Color(0xffffffff)` 
##### Venice (`VerveTheme.venice`)
* Adobe Color: [Sleep Sense](https://color.adobe.com/Sleep-Sense-color-theme-color-theme-13839732)
* `Color`s:
    - Main: `Color(0xff0396A6)`
    - Accent: `Color(0xffF25E5E)`
    - Background: `Color(0xffffffff)` 
    
### Working example
```dart
import 'package:flutter/material.dart';
import 'package:verve/verve.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    List titles = [
      'This is a list',
      'Verve adds some flair',
      'Verve can make you happy',
      'Verve has more sane theming',
      'The Metropolis font is used',
    ];
    List subtitles = [
      'made with a Verve theme.',
      'to boring old ListView.builder.',
      'if you choose to use it.',
      'defaults than other solutions.',
      'because it is nice aesthetically.',
    ];
    return MaterialApp(
      title: 'Verve Example',
      //Garden verveSwatch
      theme: verveSwatch(theme: VerveTheme.garden),
      home: Scaffold(
        appBar: AppBar(
          title: Text('Welcome to Verve!'),
        ),
        body: Center(
          child: Column(
            children: <Widget>[
              //Generator for String based ListView
              //Uses rounded, auto-styled card.
              ListView.builder(
                shrinkWrap: true,
                itemCount: titles.length,
                itemBuilder: (BuildContext context, int i) {
                  return Card(
                      child: Column(
                    children: <Widget>[
                      Container(
                          padding: EdgeInsets.fromLTRB(0.0, 5.0, 0.0, 0.0),
                          child: Text(
                            titles[i],
                            //Title theme in styled card
                            style: Theme.of(context)
                                .textTheme
                                .title
                                .copyWith(color: Colors.black),
                          )),
                      Container(
                          padding: EdgeInsets.fromLTRB(0.0, 2.5, 0.0, 5.0),
                          child: Text(
                            subtitles[i],
                            //Subtitle theme in styled card
                            style: Theme.of(context)
                                .textTheme
                                .subtitle
                                .copyWith(color: Colors.black),
                          )),
                    ],
                  ));
                },
              ),
              //Rounded-flat button.
              RaisedButton(
                  //Random callback to make sure the button theming is enabled
                  onPressed: () {
                    titles = ["x", "x"];
                  },
                  child: Text('CLICK')),
            ],
          ),
        ),
      ),
    );
  }
}

```

---
###### Verve, a Flutter design library.

###### ©2019 Aditya Kishore

###### Licensed under the Mozilla Public License 2.0


CHANGELOG.md
## [0.3.2] - [0.3.2+3] (Published 12/06/19)
* Even more bug and consistency fixes
* Bettered theming defaults
* Updated README to reflect new changes
* Package is mostly stable!

## [0.3.1] (Published 12/04/19) (Breaking!)
* More bug and consistency fixes
* Stopped defining components
* Twilight more standardized

## [0.1.1] - [0.2.1+5] (Published 12/03/19 & 12/04/19)
* Added 11 verveSwatch(es)
* Created basic set of components
* Licensed under MPL 2.0
* Wrote example and README
* Fixed bugs with components and color

example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:verve/verve.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    List titles = [
      'This is a list',
      'Verve adds some flair',
      'Verve can make you happy',
      'Verve has more sane theming',
      'The Metropolis font is used',
    ];
    List subtitles = [
      'made with a Verve theme.',
      'to boring old ListView.builder.',
      'if you choose to use it.',
      'defaults than other solutions.',
      'because it is nice aesthetically.',
    ];
    return MaterialApp(
      title: 'Verve Example',
      //Navy verveSwatch
      theme: verveSwatch(theme: VerveTheme.garden),
      home: Scaffold(
        appBar: AppBar(
          title: Text('Welcome to Verve!'),
        ),
        body: Center(
          child: Column(
            children: <Widget>[
              //Generator for String based ListView
              //Uses rounded verveCards.
              ListView.builder(
                shrinkWrap: true,
                itemCount: titles.length,
                itemBuilder: (BuildContext context, int i) {
                  return Card(
                      child: Column(
                    children: <Widget>[
                      Container(
                          padding: EdgeInsets.fromLTRB(0.0, 5.0, 0.0, 0.0),
                          child: Text(
                            titles[i],
                            style: Theme.of(context)
                                .textTheme
                                .title
                                .copyWith(color: Colors.black),
                          )),
                      Container(
                          padding: EdgeInsets.fromLTRB(0.0, 2.5, 0.0, 5.0),
                          child: Text(
                            subtitles[i],
                            style: Theme.of(context)
                                .textTheme
                                .subtitle
                                .copyWith(color: Colors.black),
                          )),
                    ],
                  ));
                },
              ),
              //Rounded-flat button.
              RaisedButton(
                  onPressed: () {
                    titles = ["x", "x"];
                  },
                  child: Text('CLICK')),
            ],
          ),
        ),
      ),
    );
  }
}


lib/verve.dart
//  This Source Code Form is subject to the terms of the Mozilla Public
//  License, v. 2.0. If a copy of the MPL was not distributed with this
//  file, You can obtain one at http://mozilla.org/MPL/2.0/.

library verve;

import 'package:flutter/material.dart';

///Default VerveTextTheme...
const VerveTextTheme = TextTheme(
  display4: TextStyle(
      fontFamily: 'Metropolis',
      fontSize: 96.0,
      fontWeight: FontWeight.w300,
      letterSpacing: -1.5,
  ),
  display3: TextStyle(
      fontFamily: 'Metropolis',
      fontSize: 60.0,
      fontWeight: FontWeight.w300,
      letterSpacing: -0.5),
  display2: TextStyle(
      fontFamily: 'Metropolis',
      fontSize: 48.0,
      fontWeight: FontWeight.w400,
      letterSpacing: 0.0),
  display1: TextStyle(
      fontFamily: 'Metropolis',
      fontSize: 34.0,
      fontWeight: FontWeight.w400,
      letterSpacing: 0.25),
  headline: TextStyle(
      fontFamily: 'Metropolis',
      fontSize: 24.0,
      fontWeight: FontWeight.w400,
      letterSpacing: 0.0),
  title: TextStyle(
      fontFamily: 'Metropolis',
      fontSize: 20.0,
      fontWeight: FontWeight.w500,
      letterSpacing: 0.15),
  subhead: TextStyle(
      fontFamily: 'Metropolis',
      fontSize: 16.0,
      fontWeight: FontWeight.w400,
      letterSpacing: 0.15),
  body2: TextStyle(
      fontFamily: 'Metropolis',
      fontSize: 14.0,
      fontWeight: FontWeight.w400,
      letterSpacing: 0.25),
  body1: TextStyle(
      fontFamily: 'Metropolis',
      fontSize: 16.0,
      fontWeight: FontWeight.w400,
      letterSpacing: 0.5),
  button: TextStyle(
      fontFamily: 'Metropolis',
      fontSize: 14.0,
      fontWeight: FontWeight.w500,
      letterSpacing: 0.75),
  caption: TextStyle(
      fontFamily: 'Metropolis',
      fontSize: 12.0,
      fontWeight: FontWeight.w400,
      letterSpacing: 0.4),
  subtitle: TextStyle(
      fontFamily: 'Metropolis',
      fontSize: 14.0,
      fontWeight: FontWeight.w500,
      letterSpacing: 0.1),
  overline: TextStyle(
      fontFamily: 'Metropolis',
      fontSize: 10.0,
      fontWeight: FontWeight.w400,
      letterSpacing: 1.5),
);

/// Uses a VerveTheme to reform a ThemeData with relevant colors.
///
/// Also updates ThemeData to use MaterialDesign 2018 specification for
/// text, as opposed to the current 2014 specification.
ThemeData verveSwatch({@required VerveTheme theme, ThemeData base}) {
  if (base == null) base = ThemeData();
  Color bg;
  Color accent;
  Color main;
  switch (theme) {
    case VerveTheme.amati:
      {
        main = Color(0xff45757D);
        accent = Color(0xff27455C);
        bg = Color(0xffffffff);
        base = base.copyWith(
          primaryColorBrightness: Brightness.dark,
          accentColorBrightness: Brightness.dark,
          brightness: Brightness.light,
        );
      }
      break;
    case VerveTheme.blue:
      {
        main = Color(0xff2787B7);
        accent = Color(0xff024669);
        bg = Color(0xffffffff);
        base = base.copyWith(
          primaryColorBrightness: Brightness.dark,
          accentColorBrightness: Brightness.dark,
          brightness: Brightness.light,
        );
      }
      break;
    case VerveTheme.purple:
      {
        main = Color(0xff73305B);
        accent = Color(0xff272640);
        bg = Color(0xffffffff);
        base = base.copyWith(
          primaryColorBrightness: Brightness.dark,
          accentColorBrightness: Brightness.dark,
          brightness: Brightness.light,
        );
      }
      break;
    case VerveTheme.venice:
      {
        main = Color(0xff0396A6);
        accent = Color(0xffF25E5E);
        bg = Color(0xffffffff);
        base = base.copyWith(
          primaryColorBrightness: Brightness.dark,
          accentColorBrightness: Brightness.dark,
          brightness: Brightness.light,
        );
      }
      break;
    case VerveTheme.twilight:
      {
        main = Color(0xff192853);
        accent = Color(0xffffffff);
        bg = Color(0xff0C1A3D);
        base = base.copyWith(
          primaryColorBrightness: Brightness.light,
          accentColorBrightness: Brightness.dark,
          brightness: Brightness.dark,
        );
      }
      break;
    case VerveTheme.beach:
      {
        main = Color(0xff046380);
        accent = Color(0xffE6E2AF);
        bg = Color(0xffffffff);
        base = base.copyWith(
          primaryColorBrightness: Brightness.light,
          accentColorBrightness: Brightness.dark,
          brightness: Brightness.light,
        );
      }
      break;
    case VerveTheme.darkMode:
      {
        return ThemeData.dark().copyWith(
          textTheme: VerveTextTheme,
        );
      }
      break;
    case VerveTheme.garden:
      {
        main = Color(0xff4BB5C1);
        accent = Color(0xffB5E655);
        bg = Color(0xffffffff);
        base = base.copyWith(
          primaryColorBrightness: Brightness.light,
          accentColorBrightness: Brightness.light,
          brightness: Brightness.light,
        );
      }
      break;
    case VerveTheme.navy:
      {
        main = Color(0xff2C3E50);
        accent = Color(0xffFC4349);
        bg = Color(0xffffffff);
        base = base.copyWith(
          primaryColorBrightness: Brightness.dark,
          accentColorBrightness: Brightness.dark,
          brightness: Brightness.light,
        );
      }
      break;
    case VerveTheme.sepia:
      {
        main = Color(0xff1A9481);
        accent = Color(0xff9BCC93);
        bg = Color(0xffffffff);
        base = base.copyWith(
          primaryColorBrightness: Brightness.dark,
          accentColorBrightness: Brightness.light,
          brightness: Brightness.light,
        );
      }
      break;
    case VerveTheme.material:
      {
        return ThemeData.light().copyWith(
          textTheme: VerveTextTheme,
        );
      }
      break;
  }
  base = base.copyWith(
      primaryColor: main,
      accentColor: accent,
      backgroundColor: bg,
      scaffoldBackgroundColor: bg,
      buttonColor: accent,
      buttonTheme: ButtonThemeData(
        buttonColor: accent,
        hoverColor: main,
        splashColor: main,
        textTheme: ButtonTextTheme.normal,
        shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(5.0)),
      ),
      cardTheme: CardTheme(
        margin: EdgeInsets.symmetric(vertical: 5, horizontal: 10.0),
        shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(7.5)),
        color: bg,
      ),
      floatingActionButtonTheme: FloatingActionButtonThemeData(
        backgroundColor: accent,
        foregroundColor: bg,
        splashColor: main,
        hoverColor: main,
      ),
      appBarTheme: AppBarTheme(brightness: base.primaryColorBrightness),
      textTheme: VerveTextTheme);
  return base;
}

///Different themes available for [verveSwatch].
enum VerveTheme {
  material,
  amati,
  venice,
  blue,
  purple,
  twilight,
  beach,
  darkMode,
  garden,
  navy,
  sepia,
}
